Classes/Common/Solr.php

```diff
@@ -247,8 +247,6 @@ public static function getSolrConnectionInfo()
         $solrInfo['password'] = $conf['solrPass'];
         // Set port if not set.
         $solrInfo['port'] = \TYPO3\CMS\Core\Utility\MathUtility::forceIntegerInRange($conf['solrPort'], 1, 65535, 8983);
-        // Append core name to path.
-        $solrInfo['path'] = trim($conf['solrPath'], '/');
```

Comment:
We still need `$solrInfo['path']`. As far as I understand the changes in Solarium, just the API path segment `/solr` is added automatically by Solarium, but if the Solr server runs in a sub-directory of the web server's document root, we still have to provide this path (although in most cases it will be `/`). So just the `solr`-part is removed from `$conf['solrPath']` and has to be added here (because this bypasses Solarium).

`$solrInfo['path'] = trim($conf['solrPath'], '/') . '/solr';`